services/src/main/java/org/keycloak/services/resources/WelcomeResource.java

```diff
@@ -269,15 +269,15 @@ private boolean shouldBootstrap() {
         return shouldBootstrap.get();
     }
 
-    private boolean isLocal() {
+    public static boolean isLocal(KeycloakSession session) {
```

Comment:
Should this rather be in `SecureContextResolver` if it's public and meant to be used beyond the Welcome page scope?

Reply:
Ok, let's leave as is. The admin root redirect is kinda related to the welcome page, so let's keep the logic there.